test/UniswapV3Staker.unit.spec.ts

```diff
@@ -612,6 +612,96 @@ describe('UniswapV3Staker.unit', async () => {
       })
     })
 
+    describe('#getRewardAmount', async () => {
+        let incentiveId: string
+        let subject: (stake: HelperTypes.Stake, incentive:  HelperTypes.Incentive) => Promise<any>
+        let timestamps: ContractParams.Timestamps
+        let stake: HelperTypes.Stake
+        let incentive: HelperTypes.Incentive
+
+        beforeEach(async () => {
+          timestamps = makeTimestamps(await blockTimestamp() + 1_000)
+
+          await erc20Helper.ensureBalancesAndApprovals(
+            lpUser0,
+            [context.token0, context.token1],
+            amountDesired,
+            context.nft.address
+          )
+
+          tokenId = await mintPosition(context.nft.connect(lpUser0), {
+            token0: context.token0.address,
+            token1: context.token1.address,
+            fee: FeeAmount.MEDIUM,
+            tickLower: getMinTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+            tickUpper: getMaxTick(TICK_SPACINGS[FeeAmount.MEDIUM]),
+            recipient: lpUser0.address,
+            amount0Desired: amountDesired,
+            amount1Desired: amountDesired,
+            amount0Min: 0,
+            amount1Min: 0,
+            deadline: (await blockTimestamp()) + 1000,
+          })
+
+          await context.nft
+            .connect(lpUser0)
+            .approve(context.staker.address, tokenId, { gasLimit: MAX_GAS_LIMIT })
+
+          await context.staker.connect(lpUser0).depositToken(tokenId)
+          const incentiveParams: HelperTypes.CreateIncentive.Args = {
+            rewardToken: context.rewardToken,
+            totalReward,
+            poolAddress: context.poolObj.address,
+            ...timestamps,
+          }
+          const stakeParams = {
+            creator: incentiveCreator.address,
+            rewardToken: context.rewardToken.address,
+            tokenId,
+            ...timestamps,
+          }
+
+          incentiveId = await helpers.getIncentiveId(
+            await helpers.createIncentiveFlow(incentiveParams)
+          )
+
+          await setTime(timestamps.startTime)
+          await context.staker.connect(lpUser0).stakeToken({
+            creator: incentiveCreator.address,
+            rewardToken: context.rewardToken.address,
+            tokenId,
+            ...timestamps,
+          })
+          stake = await context.staker.stakes(tokenId, incentiveId)
+          incentive = await context.staker.incentives(incentiveId)
+          subject = (stake: HelperTypes.Stake, incentive:  HelperTypes.Incentive) => {
+            return context.staker.connect(lpUser0).getRewardAmount(stake, incentive, stakeParams)
+          }
+      })
+
+      it('returns correct rewardAmount and secondsInPeriodX128 for the position', async () => {
+        const pool = context.poolObj.connect(actors.lpUser0())
```

Comment:
If this is only being used once, you can probably remove this line and use context.poolObj on 694